symphony-bdk-http/symphony-bdk-http-jersey2/src/main/java/com/symphony/bdk/http/jersey2/ApiClientJersey2.java

```diff
@@ -340,32 +342,59 @@ protected boolean isJsonMime(String mime) {
-        if (param.getValue() instanceof File) {
-          File file = (File) param.getValue();
-          FormDataContentDisposition contentDisp = FormDataContentDisposition.name(param.getKey())
-              .fileName(file.getName()).size(file.length()).build();
-          multiPart.bodyPart(new FormDataBodyPart(contentDisp, file, MediaType.APPLICATION_OCTET_STREAM_TYPE));
-        } else {
-          FormDataContentDisposition contentDisp = FormDataContentDisposition.name(param.getKey()).build();
-          multiPart.bodyPart(new FormDataBodyPart(contentDisp, parameterToString(param.getValue())));
+    if (contentType.startsWith(MediaType.MULTIPART_FORM_DATA)) {
+      return this.serializeMultiPartFormDataEntity(formParams);
+    }
+    else if (contentType.startsWith(MediaType.APPLICATION_FORM_URLENCODED)) {
+      final Form form = new Form();
+      formParams.forEach((key, value) -> form.param(key, parameterToString(value)));
+      return Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
+    }
+    else {
+      // We let jersey handle the serialization
+      return Entity.entity(obj, contentType);
+    }
+  }
+
+  private Entity<?> serializeMultiPartFormDataEntity(Map<String, Object> formParams) {
+
+    FormDataMultiPart multiPart = new FormDataMultiPart();
+
+    for (final Entry<String, Object> param : formParams.entrySet()) {
+      // if part is a File
+      if (param.getValue() instanceof File) {
+        final File file = (File) param.getValue();
+        final FormDataContentDisposition contentDisposition = FormDataContentDisposition
+            .name(param.getKey())
+            .fileName(file.getName())
+            .size(file.length())
+            .build();
+        final FormDataBodyPart streamPart = new FormDataBodyPart(
+            contentDisposition,
+            file,
+            MediaType.APPLICATION_OCTET_STREAM_TYPE
+        );
+        multiPart = (FormDataMultiPart) multiPart.bodyPart(streamPart);
+      }
+      // if part is a ApiClientBodyPart[]
+      else if (param.getValue() instanceof ApiClientBodyPart[]) {
+        for (ApiClientBodyPart attachment : (ApiClientBodyPart[]) param.getValue()) {
+          final StreamDataBodyPart streamPart = new StreamDataBodyPart(param.getKey(), attachment.getContent(), attachment.getFilename());
+          multiPart = (FormDataMultiPart) multiPart.bodyPart(streamPart);
         }
       }
-      entity = Entity.entity(multiPart, MediaType.MULTIPART_FORM_DATA_TYPE);
-    } else if (contentType.startsWith("application/x-www-form-urlencoded")) {
-      Form form = new Form();
-      for (Entry<String, Object> param : formParams.entrySet()) {
-        form.param(param.getKey(), parameterToString(param.getValue()));
+      // if part is a single ApiClientBodyPart
+      else if (param.getValue() instanceof ApiClientBodyPart) {
+        final ApiClientBodyPart part = (ApiClientBodyPart) param.getValue();
```

Comment:
Please extract encoding for one single ApiClientBpdyPart object

Reply:
Why for 1 single `ApiClientBodyPart` and not `ApiClientBodyPart[]`? Not really relevant to me, I'd prefer to keep it as it is